src/cephadm/cephadm

```diff
@@ -5212,6 +5261,548 @@ def command_gather_facts():
+            def wrapper(self, *args, **kwargs):
+                auth = self.headers.get("Authorization", None)
+                if auth != "Bearer " + self.server.auth_token:
+                    self.send_error(401)
+                    return
+                f(self, *args, **kwargs)
+            return wrapper
+    
+    def _help_page(self):
+        return """<!DOCTYPE html>
+<html>
+<head><title>cephadm metadata exporter</title></head>
+<style>
+body {{
+  font-family: sans-serif;
+  font-size: 0.8em;
+}}
+table {{
+  border-width: 0px;
+  border-spacing: 0px;
+  margin-left:20px;
+}}
+tr:hover {{
+  background: PowderBlue;
+}}
+td,th {{
+  padding: 5px;
+}}
+</style>
+<body>
+    <h1>cephadm metadata exporter {api_version}</h1>
+    <table>
+      <thead>
+        <tr><th>Endpoint</th><th>Methods</th><th>Response</th><th>Description</th></tr>
+      </thead>
+      <tr><td><a href='{api_version}/metadata'>{api_version}/metadata</a></td><td>GET</td><td>JSON</td><td>Return <b>all</b> metadata for the host</td></tr>
+      <tr><td><a href='{api_version}/metadata/daemons'>{api_version}/metadata/daemons</a></td><td>GET</td><td>JSON</td><td>Return daemon and systemd states for ceph daemons (ls)</td></tr>
+      <tr><td><a href='{api_version}/metadata/disks'>{api_version}/metadata/disks</a></td><td>GET</td><td>JSON</td><td>show disk inventory (ceph-volume)</td></tr>
+      <tr><td><a href='{api_version}/metadata/health'>{api_version}/metadata/health</a></td><td>GET</td><td>JSON</td><td>Show current health of the exporter sub-tasks</td></tr>
+      <tr><td><a href='{api_version}/metadata/host'>{api_version}/metadata/host</a></td><td>GET</td><td>JSON</td><td>Show host metadata (gather-facts)</td></tr>
+    </table>
+</body>
+</html>""".format(api_version=CephadmDaemonHandler.api_version)
+
+    def _fetch_root(self):
+        self.send_response(200)
+        self.send_header('Content-type', 'text/html; charset=utf-8')
+        self.end_headers()
+        self.wfile.write(self._help_page().encode('utf-8'))
+
+    @Decorators.authorize
+    def do_GET(self):
+        """Handle *all* GET requests"""
+
+        if self.path == '/':
+            # provide a html response if someone hits the root url, to document the 
+            # available api endpoints
+            return self._fetch_root()
+        elif self.path in CephadmDaemonHandler.valid_routes:
+            u = self.path.split('/')[-1]
+            data = json.dumps({})
+            if u == 'metadata':
+                data = json.dumps(self.server.cephadm_cache.to_json())
+            elif u == 'daemons':
+                data = json.dumps(self.server.cephadm_cache.daemons)
+            elif u == 'disks':
+                data = json.dumps(self.server.cephadm_cache.disks)    
+            elif u == 'health':
+                data = json.dumps(self.server.cephadm_cache.health)
+            elif u == 'host':
+                data = json.dumps(self.server.cephadm_cache.host)
+            
```

Comment:
Let's better not return a 200 when the path doesn't exist... Or am I missing something here?

```suggestion
                data = json.dumps(self.server.cephadm_cache.host)
            else:
                self.send_response(404)
            
```

Reply:
this is already in a clause of valid routes. if the route isn't valid you hit the else clause, which returns the 404 at L5409